src/util/test_util.jl

```diff
@@ -0,0 +1,197 @@
+You should run these tests if you only implement the Primary Public API -- see API section
+of docs for details about this API.
+
+These are consistency checks, not correctness tests in the absolute sense.
+For example, these tests ensure that samples generated by `rand` are of the correct size,
+but does not check that they come from the intended distribution.
+"""
+function test_finitegp_primary_public_interface(
+    rng::AbstractRNG, fx::FiniteGP; atol::Real=1e-12
+)
+    # Check that `rand` works, and produces something of the correct shape.
+    # Doesn't verify statistical properties.
+    y = rand(rng, fx)
+    @test y isa AbstractVector{<:Real}
+    @test length(y) == length(fx)
+
+    y = rand(fx)
+    @test y isa AbstractVector{<:Real}
+    @test length(y) == length(fx)
+
+    rand!(rng, fx, y)
+    rand!(fx, y)
+
+    N_samples = 3
+    Y = rand(rng, fx, N_samples)
+    @test Y isa AbstractMatrix{<:Real}
+    @test size(Y) == (length(fx), N_samples)
+
+    Y = rand(fx, N_samples)
+    @test Y isa AbstractMatrix{<:Real}
+    @test size(Y) == (length(fx), N_samples)
+
+    rand!(rng, fx, Y)
+    rand!(fx, Y)
+
+    # Ensure that `marginals` produces something of the expected size and type.
+    ms = marginals(fx)
+    @test ms isa AbstractVector{<:Normal}
+    @test length(ms) == length(fx)
+
+    # `mean`, `var`, and `mean_and_var` should be consistent with the output of marginals.
+    @test mean(fx) ≈ mean.(ms)
+    @test var(fx) ≈ var.(ms)
+    @test mean_and_var(fx)[1] ≈ mean(fx)
+    @test mean_and_var(fx)[2] ≈ var(fx)
+
+    # All elements of `var` should be positive.
+    @test all(var(fx) .> -atol)
+
+    # Ensure that `logpdf` produces a scalar.
+    @test logpdf(fx, y) isa Real
+
+    # Ensure that `posterior` produces a new AbstractGP.
+    @test posterior(fx, y) isa AbstractGP
+end
+
+"""
+    test_finitegp_primary_and_secondary_public_interface(
+        rng::AbstractRNG, fx::FiniteGP; atol::Real=1e-12
+    )
+
+Basic consistency tests for both the Primary and Secondary Public FiniteGP APIs.
+Runs `test_finitegp_primary_public_interface` as part of these tests.
+You should run these tests if you implement both the primary and secondary public APIs --
+see API section of the docs for more information about these APIs.
+
+These are consistency checks, not correctness tests in the absolute sense.
+For example, these tests ensure that samples generated by `rand` are of the correct size,
+but does not check that they come from the intended distribution.
+"""
+function test_finitegp_primary_and_secondary_public_interface(
+    rng::AbstractRNG, fx::FiniteGP; atol=1e-12
+)
+    # Test the primary API.
+    test_finitegp_primary_public_interface(rng, fx; atol=atol)
+
+    # Check that `cov` runs and is consistent with `var`.
+    @test diag(cov(fx)) ≈ var(fx)
+
+    # Check that `mean_and_cov` runs and is consistent with `mean` and `cov`.
+    @test mean_and_cov(fx)[1] ≈ mean(fx)
+    @test mean_and_cov(fx)[2] ≈ cov(fx)
```

Comment:
```suggestion
    let
        m, C = mean_and_cov(fx)
        @test m ≈ mean(fx)
        @test C ≈ cov(fx)
    end
```
as you have it below (which I think is very neat)